### PR TITLE
Blind SQL injection fix (#3284)

### PR DESCRIPTION
**Pull request by jekkos**
superseeds #3316 
sort column is now validated with the list of headers for each table.

### CODE REVIEW

File: app/Controllers/Secure_Controller.php [hunk 1]
```diff
@@ -82,6 +82,11 @@ public function __construct(string $module_id = '', string $submodule_id = null,
 		view('viewData', $this->global_view_data);
 	}
 
+	public function sortColumn($headers, $field, $default)
```
Comment (objecttothis) [inline app/Controllers/Secure_Controller.php:85]: Two thoughts:
1. I'm unclear based on the name of the function what it's doing.  It's taking in a haystack multidimensional array, string needle and default sort order then returning the needle if it's not null and exists in the flattened haystack or returning the default sort by column if not.  Our whole purpose for the function is, I think, XSS injection prevention, so I think we should choose a name for this function that reflects that.  Something like sanitizeSortByColumn() or something of that nature. 

2. I believe type hinting is the current best practice for function signatures, so I think we should add the types.  PHP has been moving more toward stricter typing and this future proofs the code.  If need be we can declare them as union types or mixed, but I think these ones are straightforward.

File: docker-compose.yml [hunk 2]
```diff
@@ -6,7 +6,7 @@ services:
         image: jekkos/opensourcepos:sql-master
         command: /bin/sh -c 'exit 0'
     ospos:
-        image: jekkos/opensourcepos:master
+        image: jekkos/opensourcepos:3.3.9
```
Comment (objecttothis) [inline docker-compose.yml:9]: Was this intended to be pushed?

**Review by objecttothis — APPROVED** [review]
@jekkos I approved this PR but only because I'm in sporadically and do not want to hold up progress if I can't re-review later.  Please take a look at my comments and decide if you want to make the recommended changes.

File: app/Controllers/Secure_Controller.php (hunk 1 above)
Comment (jekkos) [reply app/Controllers/Secure_Controller.php:85]: Agree will add the type hint before merge.

File: docker-compose.yml (hunk 2 above)
Comment (jekkos) [reply docker-compose.yml:9]: This needs to be reverted indeed

**Comment by jekkos** [comment]
comments were addressed, will merge.

**Pull request merged**